### ISSUE TITLE
Using Spectral in JavaScript produces incorrect results

### ISSUE THREAD

**Issue by disposedtrolley** [stoplightio/spectral#750]
**Describe the bug**
Importing Spectral into a NodeJS project (v8.16.1) and attempting to run the YAML example in the [docs](https://stoplight.io/p/docs/gh/stoplightio/spectral/docs/guides/javascript.md) doesn't produce the correct rule results. The docs reference a [repl.it example](https://repl.it/@ChrisMiaskowski/spectral-rules-example) which outputs 5 rule results when run, but in my program only a single rule result is produced, and the line numbers are incorrect. I've also noticed that the example uses a much older version of Spectral.

**To Reproduce**
1. Create a new `npm` project running Node v8.16.1 with a single JS file with the following contents:

```
const { Spectral } = require('@stoplight/spectral');
const { parseWithPointers } = require('@stoplight/yaml');

const myOpenApiDocument = parseWithPointers(`
responses:
  '200':
    description: ''
    schema:
      $ref: '#/definitions/error-response'
`);

const spectral = new Spectral();
spectral
  .loadRuleset('spectral:oas3') // spectral:oas2 for OAS 2.0 aka Swagger
  .then(() => spectral.run(myOpenApiDocument))
  .then(results => {
    console.log(JSON.stringify(results));
  });
```
2. Run the script with `node`
3. Observe output is as follows:

```
[{"code":"invalid-ref","path":["data","responses","200","schema","$ref"],"message":"'#/definitions/error-response' does not exist","severity":0,"range":{"start":{"line":154,"character":18},"end":{"line":154,"character":48}}}]
```

**Expected behavior**
5 rule results should be printed as shown in the repl.it example. The line numbers should be correct.

**Environment (remove any that are not applicable):**
 - Library version: 4.2.0
 - OS: macOS 10.14
 - Node: 8.16.1

**Comment by disposedtrolley** [comment]
So I did some digging and tried to emulate how the CLI calls the `spectral.run()` method. Instead of passing in the `IParserResult` value from `parseWithPointers()`, I have to wrap it in an `IParsedResult` type [like so](https://github.com/stoplightio/spectral/blob/a714874ee033e97ecabb230bc1f1e2c56c5b8062/src/cli/services/linter/linter.ts#L79)

When executed, the below examples produce a result with the correct line numbers, but still doesn't produce all 5 expected results.

```
const { Spectral } = require('@stoplight/spectral');
const {
  parseWithPointers,
  getLocationForJsonPath
} = require('@stoplight/yaml');

const myOpenApiDocument = parseWithPointers(`
responses:
  '200':
    description: ''
    schema:
      $ref: '#/definitions/error-response'
`);

console.log(myOpenApiDocument);

const spectral = new Spectral();
spectral
  .loadRuleset('spectral:oas') // spectral:oas2 for OAS 2.0 aka Swagger
  .then(() => {
    const parsedResult = {
      parsed: myOpenApiDocument,
      getLocationForJsonPath
    };
    return spectral.run(parsedResult);
  })
  .then(results => {
    console.log(JSON.stringify(results));
  });
```

**Comment by P0lip** [comment]
@disposedtrolley 
You need to register formats, like so

```ts
const { Spectral, isOpenApiv3, isOpenApiv2 } = require('@stoplight/spectral');

const spectral = new Spectral();
spectral.registerFormat('oas2', isOpenApiv2);
spectral.registerFormat('oas3', isOpenApiv3);
```
and then obviously `openapi` or `swagger` needs to be specified in the document so that the format can be matched, i.e
```
const myOpenApiDocument = parseWithPointers(`
openapi: 3.0.0
responses:
  '200':
    description: ''
    schema:
      $ref: '#/definitions/error-response'
`);
```

The repl we have included it ancient. AFAIR I've updated docs recently, but they might have not been deployed yet.

**Comment by disposedtrolley** [comment]
Thanks @P0lip for the quick response!

I tried your solution and I'm running into issues. `isOpenApiv3` is undefined when required from `@stoplight/spectral`, so the `registerFormat()` method fails with:

    (node:55070) UnhandledPromiseRejectionWarning: TypeError: this.formats[format] is not a function

It looks like the entire [formats module](https://github.com/stoplightio/spectral/tree/develop/src/formats) is missing from the `dist` folder:

<img width="372" alt="image" src="https://user-images.githubusercontent.com/4476568/68434474-1f3ad080-020d-11ea-9491-8c211166e037.png">

**Comment by disposedtrolley** [comment]
Nevermind, works fine after removing and reinstalling node modules!
